torchvision/prototype/transforms/_geometry.py

```diff
@@ -414,3 +422,201 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
             fill=self.fill,
             center=self.center,
         )
+
+
+class RandomCrop(Transform):
+    def __init__(
+        self,
+        size: Union[int, Sequence[int]],
+        padding: Optional[Union[int, Sequence[int]]] = None,
+        pad_if_needed: bool = False,
+        fill: Union[int, float, Sequence[int], Sequence[float]] = 0,
+        padding_mode: Literal["constant", "edge", "reflect", "symmetric"] = "constant",
+    ) -> None:
+        super().__init__()
+
+        self.size = _setup_size(size, error_msg="Please provide only two dimensions (h, w) for size.")
+
+        if padding is not None:
+            _check_padding_arg(padding)
+
+        if (padding is not None) or pad_if_needed:
+            _check_padding_mode_arg(padding_mode)
+            _check_fill_arg(fill)
+
+        self.padding = padding
+        self.pad_if_needed = pad_if_needed
+        self.fill = fill
+        self.padding_mode = padding_mode
+
+    def _get_params(self, sample: Any) -> Dict[str, Any]:
+        image = query_image(sample)
+        _, height, width = get_image_dimensions(image)
+        output_height, output_width = self.size
+
+        if height + 1 < output_height or width + 1 < output_width:
+            raise ValueError(
+                f"Required crop size {(output_height, output_width)} is larger then input image size {(height, width)}"
+            )
+
+        if width == output_width and height == output_height:
+            return dict(top=0, left=0, height=height, width=width)
+
+        top = torch.randint(0, height - output_height + 1, size=(1,)).item()
+        left = torch.randint(0, width - output_width + 1, size=(1,)).item()
+        return dict(top=top, left=left, height=output_height, width=output_width)
+
+    def _forward(self, flat_inputs: List[Any]) -> List[Any]:
```

Comment:
My understanding is that `_forward` receives the whole flattened input while `_transform` would require a single input. But why do we need another new idiom here? We should avoid introducing multiple idioms on how to structure the Transforms if we can.

Can we move some of the param estimations done here on the `_get_params()` and use the rewrite this to do all ops in a `_transform` individually?

Reply:
There can be 2 options:
-    We acknowledge that this Transform is different from the rest and overwrite the forward(). What you proposed above looks OK to me though it hides information on what it does and it's harder to read.
-    We try to estimate the sizes on get_params and then you put the whole implementation in _transform() as per all other transforms. (we still need to make sure if it feasible)

The option 1 is almost done and shows a way how to do things in a mixed way: _transform + other things.

Option 2, will require to create mid-level and low-level ops doing padded crop op (same as resized_crop) but this way (I have to verify that though if feasible) we keep following our main paradigm (implementing only _transform) with RandomCrop transform. A minor drawback of option2 is that we implictly show by our code to the users that a composed ops should be done at mid/low-level.
While option 1 shows that Transforms could be easily overridden and "hacked" as user wants ...